### PR TITLE
PORTALS-2229

### PR DESCRIPTION
**Pull request by jay-hodgson**

### CODE REVIEW

File: src/_ToggleSynapseObjects.scss [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// Import the theme css
+@use 'variables' as Portal;
+
+.ToggleSynapseObjects {
+  margin-top: 30px;
+  .reactSwitchContainer{
+    width: 60px;
+    position: absolute;
+    right: 0px;
+    margin-top: -54px;
```
Comment (jay-hodgson) [inline src/_ToggleSynapseObjects.scss:10]: Lock it to the right side via right: 0px, and move it up via a negative margin-top.

File: src/_ToggleSynapseObjects.scss [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  }
+  color: Portal.$primary-action-color;
```
Comment (jay-hodgson) [inline src/_ToggleSynapseObjects.scss:12]: Use the themed primary action color (for the selected view), but override with a light grey when unselected.

File: package.json [hunk 3]
```diff
@@ -16,6 +16,7 @@
     "react-dom": "16.14.0",
     "react-router-dom": "5.3.3",
     "react-spinners": "^0.11.0",
+    "react-switch": "^7.0.0",
```
Comment (jay-hodgson) [inline package.json:19]: So happy to find this.  https://www.npmjs.com/package/react-switch

File: src/configurations/adknowledgeportal/synapseConfigs/experimental_tools.ts [hunk 4]
```diff
@@ -61,47 +61,49 @@ export const experimentalToolsCardConfiguration: CardConfiguration = {
   ],
 }
 
+export const experimentalDetailsTableConfigurationColumnLinks: LabelLinkConfig = [
```
Comment (jay-hodgson) [inline src/configurations/adknowledgeportal/synapseConfigs/experimental_tools.ts:64]: Refactored for reuse in the Project details page (where it shows the associated experimental models table)

File: src/configurations/adknowledgeportal/synapseConfigs/projects.ts [hunk 5]
```diff
@@ -68,15 +67,35 @@ export const projectsDetailsPageConfiguration: DetailsPageProps = {
...
-      columnName: 'Grant Number',
+      name: 'ToggleSynapseObjects',
       title: 'Experimental Models',
       showTitleSeperator: false,
-      tableSqlKeys: ['grant'],
-      props: {
-        sql: experimentalModelsSql,
-        ...experimentalToolsCardConfiguration,
+      standalone: true,
+      toggleConfigs: {
```
Comment (jay-hodgson) [inline src/configurations/adknowledgeportal/synapseConfigs/projects.ts:74]: toggleConfigs defines the two configs to use, along with what icons (Icon enum defined in IconSvg)

File: src/portal-components/DetailsPage.tsx [hunk 6]
```diff
@@ -438,6 +438,15 @@ function isReactFragment(variableToInspect: any): boolean {
   return variableToInspect === React.Fragment
 }
 
+const getSynapseComponent = (el: RowSynapseConfig, queryResultBundle?: QueryResultBundle) => {
```
Comment (jay-hodgson) [inline src/portal-components/DetailsPage.tsx:441]: Pulled out existing code into a reusable function, that will also be called when processing toggleConfigs

File: src/portal-components/DetailsPage.tsx [hunk 7]
```diff
@@ -448,7 +457,7 @@ export const DetailsPageSynapseConfigArray: React.FC<{
     <>
       {synapseConfigArray.map((el: RowSynapseConfig, index) => {
         const id = COMPONENT_ID_PREFIX + index
-        const { standalone, resolveSynId, showTitleSeperator = true } = el
+        const { resolveSynId, showTitleSeperator = true } = el
```
Comment (jay-hodgson) [inline src/portal-components/DetailsPage.tsx:460]: Passes entire el (RowSynapseConfig) to getSynapseComponent(), so no need to pull out the standalone value

**Comment by jay-hodgson** [comment]
Demo video shown in the associated jira ticket: https://sagebionetworks.jira.com/browse/PORTALS-2229?focusedCommentId=149688

File: src/configurations/adknowledgeportal/synapseConfigs/projects.ts [hunk 8]
```diff
@@ -68,15 +67,35 @@ export const projectsDetailsPageConfiguration: DetailsPageProps = {
...
+          props: {
+            sql: experimentalModelsSql,
+            rgbIndex,
+            sqlOperator: '=',
+            columnLinks: experimentalDetailsTableConfigurationColumnLinks,
+          },
+          columnName: 'Grant Number',
+          tableSqlKeys: ['grant'],
+        },
+        icon2: 'entityview',
```
Comment (jay-hodgson) [inline src/configurations/adknowledgeportal/synapseConfigs/projects.ts:87]: Need to update this icon before merge (to 'cards', which will be available in the next SRC release)

**Review by nickgros — APPROVED** [review]

**Pull request merged**